plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitManager.java

```diff
@@ -71,8 +80,21 @@ public ConnectorSplitSource getSplits(
 
         // TODO Use residual. Right now there is no way to propagate residual to Trino but at least we can
         //      propagate it at split level so the parquet pushdown can leverage it.
-        IcebergSplitSource splitSource = new IcebergSplitSource(tableScan.planTasks());
+        ImmutableList.Builder<FileScanTask> fileScanTasks = ImmutableList.builder();
```

Comment:
This change makes splits materialized eagerly during planning. While previously (IIUC) they were listed lazilly as the execution went. 

I would prefer to not change that and fail only if we encounter the file which contains deletes as we go.

It may be later during the query, which is a somewhat worse experience for the user when they try to read V2 table with deletes. But I would argue that we should not optimize for nicer handling of unsupported tables if that hinders the typical case.

Reply:
I agree with this. Iterating through the result of `planTasks` will perform planning separately just to validate there are no delete files, which is almost certainly something you don't want to do.

Reply:
Gotcha, thanks. I didn't have a good sense of how much work was being done lazily in the `CloseableIterable<CombinedScanTask>`. I'll switch it back.